test/tests/kernels/2d_diffusion/tests

```diff
@@ -5,27 +5,48 @@
     exodiff = 'out.e'
     scale_refine = 5
     group = 'requirements'
+    requirement = 'MOOSE shall test Dirichlet BC for a 2D diffusion problem.'
```

Comment:
The requirements shouldn't be that things should be tested -- they should be about the things that are tested:

MOOSE shall provide an ability to solve a 2D diffusion problem with Dirichlet boundary conditions.